app/models/attachment_reference.rb

```diff
@@ -0,0 +1,61 @@
+class AttachmentReference < ActiveRecord::Base
+  after_initialize :set_expiry_at, if: proc { new_record? && !attachable }
+  before_validation :update_expiry_at
```

Comment:
@lowjoel I didn't implement the validation for this class, because the record will always be valid after `before_validation`.

And this callback is needed because sometimes the attachable is directly updated without calling `attachable=` (e.g. if we create a material with an attachment_reference and save the material).